### PR TITLE
Auth: Introduce Github OAuth2

### PR DESCRIPTION
**Pull request by kamoltat**
Enables the user to sign in using Github OAuth2,
using Beaker Middleware as a session plugin.
Only users that are in the Ceph community can
sign in and (In the future) have the priviledge to kill runs
or reschedule them and etc.

Signed-off-by: Kamoltat Sirivadhna <ksirivad@redhat.com>

### CODE REVIEW

**Comment by kshtsk** [comment]
I was trying to deploy this, and get while starting pulpito:
```Traceback (most recent call last):
  File "run.py", line 10, in <module>
    simpleapp_wsgi_app = deploy(prod_config)
  File "/home/pulpito/pulpito/virtualenv/lib/python2.7/site-packages/pecan/deploy.py", line 9, in deploy
    return load_app(config)
  File "/home/pulpito/pulpito/virtualenv/lib/python2.7/site-packages/pecan/core.py", line 218, in load_app
    app = module.app.setup_app(_runtime_conf, **kwargs)
  File "/home/pulpito/pulpito/pulpito/app.py", line 15, in setup_app
    **app_conf
  File "/home/pulpito/pulpito/virtualenv/lib/python2.7/site-packages/pecan/__init__.py", line 86, in make_app
    app = Pecan(root, **kw)
  File "/home/pulpito/pulpito/virtualenv/lib/python2.7/site-packages/pecan/core.py", line 844, in __init__
    super(Pecan, self).__init__(*args, **kw)
  File "/home/pulpito/pulpito/virtualenv/lib/python2.7/site-packages/pecan/core.py", line 240, in __init__
    root = self.__translate_root__(root)
  File "/home/pulpito/pulpito/virtualenv/lib/python2.7/site-packages/pecan/core.py", line 275, in __translate_root__
    module = __import__(name, fromlist=fromlist)
  File "/home/pulpito/pulpito/pulpito/controllers/root.py", line 16, in <module>
    from pulpito.controllers.login import LoginController
  File "/home/pulpito/pulpito/pulpito/controllers/login.py", line 2, in <module>
    from requests_oauthlib import OAuth2Session
ImportError: No module named requests_oauthlib
```

**Comment by kshtsk** [comment]
I guess you need to add `requests-oauthlib==1.3.0` to requirements.

**Review by kshtsk — CHANGES_REQUESTED** [review]
See comments.

**Comment by kshtsk** [comment]
If I add those dependencies the pulpito is deploying okay. On the main page I can see "Login" linke, once I click it redirects me to the GitHub 404 error page and url: https://github.com/login/oauth/authorize?response_type=code&client_id=None&state=rhO2YihSf8PlloOXMAtsXWN4YGiDbo

**Comment by kshtsk** [comment]
Rebase required

**Comment by kamoltat** [comment]
@kshtsk you might have to include a .env file which has CLIENT_ID and CLIENT_SECRET. I'll send the file over to you through email

**Comment by kamoltat** [comment]
> Rebase required

done!

**Comment by kshtsk** [comment]
need to rebase

File: pulpito/app.py [hunk 1]
```diff
@@ -1,16 +1,18 @@
-from pecan import conf, make_app
+from pecan import conf, make_app, request
```
Comment (zmc) [inline pulpito/app.py:1]: Is this a mistake?

File: pulpito/controllers/login.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+from requests_oauthlib import OAuth2Session
+import urlparse
+import os
+import json
+from dotenv import load_dotenv, find_dotenv
+from pulpito.controllers import session
+
+
+# TODO: Uncomment for development without https
+os.environ['OAUTHLIB_INSECURE_TRANSPORT'] = '1'
```
Comment (zmc) [inline pulpito/controllers/login.py:11]: Could this be done with the `.env` perhaps?

**Review by zmc — APPROVED** [review]
Aside from some very minor comments, I think this looks great.

**Comment by kamoltat** [comment]
Closing this PR, please refer to: https://github.com/VallariAg/teuthology-api/pull/5

**Comment by kamoltat** [comment]
Closing this PR, please refer to: https://github.com/VallariAg/teuthology-api/pull/5

**Pull request closed without merging**